lib/api/api_client.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:pretty_dio_logger/pretty_dio_logger.dart';
+
+typedef AccessTokenProvider = Future<String> Function();
+
+class ApiClient {
+  ApiClient({required this.baseUrl}) : assert(baseUrl != null) {
+    dio = Dio()
+      ..options.baseUrl = baseUrl!
+      ..interceptors.add(InterceptorsWrapper(onRequest: (options, handler) => handler.next(options)));
```

Comment:
add `onResponse` and `onError` properties

`onResponse: (response, handler) => handler.next(response)`
`onError: (DioError e, handler) => handler.next(e)`